MapleServer2/Packets/QuestPacket.cs

```diff
@@ -18,23 +20,23 @@ private enum QuestType : byte
 
         // Client has some sort of caching for this packet resulting subsequent packets to not change anything
         // TODO: Find an way to overwrite the cache in the client.
-        public static Packet SendQuests(List<QuestMetadata> questList)
+        public static Packet SendQuests(List<QuestStatus> questList)
         {
             PacketWriter pWriter = PacketWriter.Of(SendOp.QUEST);
             pWriter.WriteEnum(QuestType.SendQuests);
-            pWriter.WriteInt(questList.Count);
 
-            foreach (QuestMetadata item in questList)
+            pWriter.WriteInt(questList.Count);
+            foreach (QuestStatus quest in questList)
             {
-                pWriter.WriteInt(item.Basic.QuestID);
-                pWriter.WriteInt(0); // 0 and 0 to set to not started, 1 and 0 to started not completed, use 2 and 1 to set to completed
-                pWriter.WriteInt(0);
-                pWriter.WriteLong(DateTimeOffset.UtcNow.ToUnixTimeSeconds()); // Timestamp of receiving the quest
-                pWriter.WriteLong(); // Timestamp of completion
-                pWriter.WriteByte(item.Basic.AutoStart); // unsure need more testing
-                pWriter.WriteInt(item.Condition.Count);
+                pWriter.WriteInt(quest.Basic.QuestID);
+                pWriter.WriteInt(quest.Completed ? 2 : quest.Started ? 1 : 0); // 0 and 0 to set to not started, 1 and 0 to started not completed, use 2 and 1 to set to completed
+                pWriter.WriteInt(quest.Completed ? 1 : 0);
```

Comment:
This makes the code longer but much more readable and it documents itself (don't need that long comment)
```
    if (quest.Completed) {
        pWriteInt(2);
        pWriteInt(1);
    } else if (quest.Started) {
        pWriteInt(1);
        pWriteInt(0);
    } else {
        pWriteInt(0);
        pWriteInt(0);
    }
```